Client/Frontend/ContentBlocker/ContentBlockerHelper.swift

```diff
@@ -160,14 +161,23 @@ class ContentBlockerHelper {
         }
     }
 
-    private func addActiveRulesToTab() {
+    private var isTPInstalledForTab = false
+
+    // Function to install or remove TP for a tab, if a main doc URL is available on the tab, it can be checked for whitelisting.
+    func setupTabTrackingProtection(forUrl url: URL?) {
         if !ContentBlockerHelper.heavyInitHasRunOnce {
             return
         }
 
+        let isInstalling = isEnabled && (url != nil && !ContentBlockerHelper.isWhitelisted(url: url!))
```

Comment:
force unwrapping 😢

Reply:
This variable is called `isInstalling` but it is also checking if the site is whitelisted. we should separate this so its clear whats happening here.